### ISSUE TITLE
feature: support auto-cloning of source control backed services

### ISSUE THREAD

**Issue by mpareja** [meet-tarantino/tarantino#7]
We like that Tarantino can clone your source control repositories on the fly. We also like that Tarantino doesn't try and clone repositories for services that aren't source control backed (like mongo, redis, etc.). 

We have a few responsibilities that need to be filled one way or another, implicitly or explicitly:

  + know which services are SCM backed (and which are not)
  + know the URL to the source control repository for each service
  + know the type of source control repository (git, svn, mercurial)
 
Our original Tarantino model assumed a few things about source control:

  1. all repositories use Git
  2. all repositories belong to one GitHub organization
  3. all repository names precisely match `docker-compose` service names

We need to decide how to add this functionality. We need to be cognizant that changing how we fulfill the set of responsibilities noted above will impact Tarantino users - even if that's only 360 for now.

**Comment by mpareja** [comment]
I'm spiking out what it would look like to use the `docker-compose.yml` file's ability to label things....

**Comment by darrenmce** [comment]
i think i tried that, but maybe you can find a loophole :)

i ran into 2 issues:

  1) parsing the yaml stinks, would rather rely less on it
  2) adding meta to the docker container requires it to be running to scrape from an `inspect`

**Comment by darrenmce** [comment]
fyi: https://docs.docker.com/compose/compose-file/#/labels-1 was what i was trying to leverage

**Comment by mpareja** [comment]
Okay, I spiked out and working in PR #9. Yup, YAML in bash stinks. I went with it mostly because it would be nice not to have a separate file where we need to map `service name` => `some service-specific setting` (ex: `bob` => `source: meet-tarantino/bob`). It would have been possible to invert the relationship and go `boolean setting` => `services with it on` (for instance, `source backed services` => `alice bob sally`) but I don't think there is much runway to that approach.
